### DISCUSSION TITLE
Threshold Key (TKey) usage in Flutter SDK and tKey + social login token expiry

### DISCUSSION THREAD

**Discussion by asadahmedtech** [Web3Auth/Web3Auth#1190] [Help]
Is there a possibility of getting the torus key independently in application using flutter/kotlin/swift SDK, similar to https://web3auth.io/docs/sdk/self-host/usage this example. I need the tKey which is generated at the application level. 

Also, since tKey is created using social login, the OAuth token will have an expiry associated with. How is the OAuth token expiry and refresh handled at torus node level.

**Comment by shahbaz17** [comment]
Hey @asadahmedtech 

Currently, the self-host is available only on the Web. For mobile, works is in development.

To your second question, tKey would need idToken each time a login is initiated. And you would need to manage the session on your end.

**Comment by asadahmedtech** [comment]
Hey @shahbaz17,
In case of session management, if we go by auth0 based auth there will be a default expiry. How do you suggest we handle expiry of token, will web3auth automatically refresh the validity in torus nodes or the tkey created will no longer be accessible? 
Just wanted to get an idea how to make sure the tKey created is not hampered for the user
